servercom/servercom.go

```diff
@@ -0,0 +1,138 @@
+package serverCom
+
+import (
+	"fmt"
+	mm "github.com/mattermost/platform/model"
+)
+/*
+ServerCom acts as a mitigator between the frontend and the mattermost model API.
+*/
+type ServerCom struct {
+	Client mm.Client
+	Team mm.Team
+	Channel *mm.Channel
+}
+
+/*
+Startup accepts the url and login credentials for a user, and returns a new serverCom struct.
+*/
+func Startup(url string, username string, password string) ServerCom {
```

Comment:
@aThorp96 @camwood23 [This code](https://github.com/mattermost/mattermost-bot-sample-golang/blob/master/bot_sample.go#L82) shows how to detect events on the server. Not sure how much adaptation it might need to work for us, but it's a better method than polling.